docs/conceptual/runtime/contracts/contracts_module.md

```diff
@@ -0,0 +1,125 @@
+module on your blockchain.
+
+### Runtime Environment Types
+
+For writing contracts and interacting with the runtime, a set of types are available (e.g.
+`AccountId`, `Balance`, `Hash`, `Moment`). These types can be user defined for custom runtimes, or
+the supplied defaults can be used.
+
+### Contract Calls
+
+Calls to contracts can alter the storage of the contract, create new contracts, and call other
+contracts. Because Substrate provides you with the ability to write custom runtime modules, the
+Contracts module also enables you to make asynchronous calls directly to those runtime functions on
+behalf of the contract's account.
+
+### Sandboxed
+
+The Contracts module is intended to be used by any user on a public network. This means that
+contracts only have the ability to directly modify their own storage. To provide safety to the
+underlying blockchain state, the Contracts module enables revertible transactions, which roll back
+any changes to the storage by contract calls that do not complete successfully.
+
+### Gas
+
+Contract calls are charged a gas fee to limit the amount of computational resources a transaction
+can use. When forming a contract transaction, a gas limit is specified. As the contract executes,
+gas is incrementally used up depending on the complexity of the computation. If the gas limit is
+reached before the contract execution completes, the transaction fails, contract storage is
+reverted, and the gas fee is **not** returned to the user. If the contract execution completes with
+remaining gas, it is returned to the user at the end of the transaction.
+
+The Contracts module determines the gas price, which is a conversion between the Substrate
+`Currency` and a single unit of gas. Thus, to execute a transaction, a user must have a free balance
+of at least `gas price` * `gas limit` which can be spent.
+
+### Storage Rent
+
+Similar to how gas limits the amount of computational resources that can be used during a
+transaction, storage rent limits the footprint that a contract can have on the blockchain's storage.
+A contract account is charged proportionally to the amount of storage its account uses. When a
+contract's balance goes below a defined limit, the contract's account is turned into a "tombstone"
+and its storage is cleaned up. A tombstone contract can be restored by providing the data that was
+cleaned up when it became a tombstone as well as any additional funds needed to keep the contract
+alive.
+
+## Contracts Module vs EVM
+
+The SRML Contracts module iterates on existing ideas in the smart contract ecosystem, particularly
+Ethereum and the EVM.
+
+The most obvious difference between the Contracts module and the EVM is the underlying execution
+engine used to run smart contracts. The EVM is a good theoretical execution environment, but it is
+not very practical use with modern hardware. For example, manipulation of 256 bit integers on modern
```

Comment:
```suggestion
not very practical to use with modern hardware. For example, manipulation of 256 bit integers on modern
```
I'm not sure if adding `to` or `for` or deleting `use` and adding `when used` makes the most sense, but the current sentence reads like: "The EVM is not very practical use with modern hardware", so something should change.